### PR TITLE
Add filtering against foreignkey fields 

### LINKED ISSUE #83: Filtering against FK field
We should be able to filter against a foreign key field.
Example:

We have the datamodel

Book
- author (fk)
- published_date (date)
- editor (txt)
- ...

Author 
- first_name (txt)
- last_name (txt)
- city (txt)

We should be able to do these kind of filters:

- /api/v1.1/book/?author__first_name=John
- /api/v1.1/book/?author__city__in=paris,london
- ...

### PR DESCRIPTION
**Pull request by Souleymane-T**
Fix #83

### CODE REVIEW

**Review by KhaledBousrih — APPROVED** [review]

File: concrete_datastore/api/v1/filters.py [hunk 1]
```diff
@@ -22,6 +20,48 @@
...
+            .get_internal_type()
+        )
+
+    # If we have field__fkfield__fkfieldfkfield or more, raise an error
+    # We can at most have field__fkfield as filter
+    else:
+        raise ValidationError(
+            {
+                "message": (
+                    f"{param}: filter not available for more than 2 levels"
```
Comment (lcognat) [inline concrete_datastore/api/v1/filters.py:59]: if there is a constraints on a 2-dimension after split, that mean a "1 level"... No ?

File: concrete_datastore/api/v1/filters.py [hunk 2]
```diff
@@ -22,6 +20,48 @@
 )
 
 
+def get_filter_field_type(model_class, param):
+    """
+    Return the type of the target field that we want to filter
```
Comment (lcognat) [inline concrete_datastore/api/v1/filters.py:25]: "The return type is a String" would be nice as an aditionnal comment

File: concrete_datastore/api/v1/filters.py [hunk 3]
```diff
@@ -468,9 +502,9 @@ def get_custom_filters(param):
             if target_field not in filterset_fields:
                 return None
 
-            target_field_type = queryset.model._meta.get_field(
-                target_field
-            ).get_internal_type()
+            target_field_type = get_filter_field_type(
+                queryset.model, target_field
+            )
             if target_field_type not in RANGEABLE_TYPES:
```
Comment (lcognat) [inline concrete_datastore/api/v1/filters.py:508]: Did you ensure `RANGEABLE_TYPES` is a list of `str` ?

**Review by lcognat — CHANGES_REQUESTED** [review]

File: concrete_datastore/api/v1/filters.py (hunk 2 above)
Comment (Souleymane-T) [reply concrete_datastore/api/v1/filters.py]: Fixed thank you

File: concrete_datastore/api/v1/filters.py (hunk 1 above)
Comment (Souleymane-T) [reply concrete_datastore/api/v1/filters.py]: Yes it is one level, not two

File: concrete_datastore/api/v1/filters.py (hunk 3 above)
Comment (Souleymane-T) [reply concrete_datastore/api/v1/filters.py:529]: No but it's define above as this:
```python
RANGEABLE_TYPES = (
    'DateTimeField',
    'DateField',
    'DecimalField',
    'IntegerField',
    'FloatField',
)
```

**Review by lcognat — APPROVED** [review]

**Pull request merged**